src/mame/heathkit/h89.cpp

```diff
@@ -96,7 +100,8 @@ class h89_state : public driver_device
 	required_device<ram_device> m_ram;
 	required_shared_ptr<uint8_t> m_floppy_ram;
 	required_device<heath_tlb_connector> m_tlbc;
-	required_device<heath_z37_fdc_device> m_h37;
+	optional_device<heath_z37_fdc_device> m_h37;
+	optional_device<heath_h_88_cass_device> m_cassette;
```

Comment:
Can you use derived classes for the machines with different capabilities and avoid `optional_device`?